kafka/producer/base.py

```diff
@@ -35,7 +38,6 @@ def _send_upstream(queue, client, codec, batch_time, batch_size,
     a specified timeout and send them upstream to the brokers in one
     request
     """
-    stop = False
```

Comment:
Looks like you removed an unused variable. Don't.

The business of forking is to minimize the delta, so that we only add what's absolutely necessary for the feature we want.